src/core/argument_parser.cpp

```diff
@@ -422,12 +438,27 @@ gs::args::parse_args_and_params(int argc, const char* const argv[]) {
         std::exit(0);
     }
 
-    auto opt_params_result = gs::param::read_optim_params_from_json(params->optimization.strategy);
-    if (!opt_params_result) {
-        return std::unexpected(std::format("Failed to load optimization parameters: {}",
-                                           opt_params_result.error()));
+    if (config_file != "") {
```

Comment:
Given that you implement the comment below (assemble name before calling) you can remove this if-else clause.